opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -77,3 +90,116 @@ def export(self, metrics: Sequence[Metric]) -> MetricExportResult:
 
     def shutdown(self) -> None:
         pass
+
+
+class PeriodicExportingMetricReader(MetricReader):
+    """`PeriodicExportingMetricReader` is an implementation of `MetricReader`
+    that collects metrics based on a user-configurable time interval, and passes the
+    metrics to the configured exporter.
+    """
+
+    def __init__(
+        self,
+        exporter: MetricExporter,
+        export_interval_millis: Optional[float] = None,
+        export_timeout_millis: Optional[float] = None,
+    ) -> None:
+        self._exporter = exporter
+        if export_interval_millis is None:
+            try:
+                export_interval_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_INTERVAL", 60000)
+                )
+            except ValueError:
+                export_interval_millis = 60000
+        if export_timeout_millis is None:
+            try:
+                export_timeout_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_TIMEOUT", 30000)
+                )
+            except ValueError:
+                export_timeout_millis = 30000
+        self._export_interval_millis = export_interval_millis
+        self._export_timeout_millis = export_timeout_millis
+        self._meter_provider = None  # type: MeterProvider
+        self._shutdown = False
+        self._daemon_thread = threading.Thread(
+            target=self.collect, daemon=True
+        )
+        self._condition = threading.Condition()
+        self._flush_event = None  # type: Optional[threading.Event]
+        self._lock = threading.Lock()
+        self._daemon_thread.start()
+        if hasattr(os, "register_at_fork"):
```

Comment:
```suggestion
        if version_info >= (3, 7):
```
It is a bit unusual to check for attributes like this. Checking for version has the advantage that it makes it obvious that this check is there only because a previous version does not have a particular feature and it makes it easy to find it and remove it when the last version that does not have it is no longer supported in this project.